treasuries/src/payout.rs

```diff
@@ -0,0 +1,54 @@
+use core::fmt::Debug;
+use frame_support::traits::tokens::PaymentStatus;
+use scale_info::TypeInfo;
+use sp_runtime::{
+	codec::{FullCodec, MaxEncodedLen},
+	DispatchError,
+};
+
+/// Can be implemented by `PayFromAccount` using a `fungible` impl, but can also be implemented with
+/// XCM/Asset and made generic over assets.
+///
+/// It is similar to the `frame_support::traits::tokens::Pay`, but it offers a variable source
+/// account for the payment.
+pub trait Payout {
+	/// The type by which we measure units of the currency in which we make payments.
+	type Balance;
+	/// AccountId
+	type AccountId;
+
+	/// The type for the kinds of asset that are going to be paid.
+	///
+	/// The unit type can be used here to indicate there's only one kind of asset to do payments
+	/// with. When implementing, it should be clear from the context what that asset is.
+	type AssetKind;
+	/// An identifier given to an individual payment.
+	type Id: FullCodec + MaxEncodedLen + TypeInfo + Clone + Eq + PartialEq + Debug + Copy;
+	/// An error which could be returned by the Pay type
+	type Error: Debug + Into<DispatchError>;
+	/// Make a payment and return an identifier for later evaluation of success in some off-chain
+	/// mechanism (likely an event, but possibly not on this chain).
+	fn pay(
+		from: &Self::AccountId,
+		to: &Self::AccountId,
+		asset_kind: Self::AssetKind,
+		amount: Self::Balance,
+	) -> Result<Self::Id, Self::Error>;
+
+	fn is_asset_supported(asset_id: &Self::AssetKind) -> bool;
```

Comment:
Not sure if we need this. We can just limit the asset id enum to supported values.